src/lib/world/src/importing.rs

```diff
@@ -1,172 +1,221 @@
-/// is a file, if the region folder exists, if the region folder is a file, and if the region folder
-/// is empty.
-fn check_paths_validity(import_dir: PathBuf) -> Result<(), WorldError> {
-    if !import_dir.exists() {
-        error!(
-            "Import path does not exist: {}",
-            import_dir.better_display()
-        );
-        return Err(WorldError::InvalidImportPath(import_dir.better_display()));
-    }
-    if import_dir.is_file() {
-        error!("Import path is a file: {}", import_dir.better_display());
-        return Err(WorldError::InvalidImportPath(import_dir.better_display()));
-    }
+/// TODO: dynamically find the best according to the system
+const BATCH_SIZE: usize = 1000; // Number of chunks to process before flushing
+const MAX_CONCURRENT_TASKS: usize = 512; // Limit concurrent tasks to prevent memory issues
+const FLUSH_INTERVAL: u64 = 10_000; // Flush every 10,000 chunks
 
-    if let Ok(dir) = import_dir.read_dir() {
-        if dir.count() == 0 {
-            error!(
-                "Import path's region folder is empty: {}",
-                import_dir.better_display()
-            );
-            return Err(WorldError::NoRegionFiles);
+impl World {
+    async fn process_chunk_batch(
+        &self,
+        chunks: Vec<VanillaChunk>,
+        progress: Arc<ProgressBar>,
+        processed_since_flush: Arc<AtomicU64>,
+    ) -> Result<(), WorldError> {
+        let chunk_objects: Vec<Chunk> = chunks
+            .into_iter()
+            .filter_map(|chunk| chunk.to_custom_format().ok())
+            .collect();
+
+        let mut success_count = 0;
+        if let Ok(()) = save_chunk_internal_batch(self, chunk_objects.clone()).await {
+            success_count = chunk_objects.len();
         }
-    } else {
-        error!(
-            "Could not read import path's region folder: {}",
-            import_dir.better_display()
-        );
-        return Err(WorldError::InvalidImportPath(import_dir.better_display()));
+
+        progress.inc(success_count.try_into().unwrap());
+
+        let total_processed = processed_since_flush
+            .fetch_add(success_count as u64, Ordering::Relaxed)
+            + success_count as u64;
+        if total_processed >= FLUSH_INTERVAL {
+            self.storage_backend.flush().await?;
+            processed_since_flush.store(0, Ordering::Relaxed);
+            info!("Performed periodic flush after {} chunks", total_processed);
+        }
+
+        Ok(())
     }
-    Ok(())
-}
 
-impl World {
-    fn get_chunk_count(&self, import_dir: PathBuf) -> Result<u64, WorldError> {
+    fn get_chunk_count(&self, import_dir: &Path) -> Result<u64, WorldError> {
         info!("Counting chunks in import directory...");
         let regions_dir = import_dir.join("region").read_dir()?;
         let chunk_count = AtomicU64::new(0);
+
         regions_dir
-            .into_iter()
             .par_bridge()
-            .for_each(|region_file| match region_file {
```

Comment:
if you are removing these checks, make sure it still works if there isn't valid data to import and that there are clear messages if the data isn't valid

Reply:
should not be a problem